Port remaining docs to JSDoc format

gregkare opened the pull request:
Connected to #1006

skddc commented on src/remotestorage.js:
```diff
@@ -682,7 +667,16 @@ RemoteStorage.prototype = {
+   * sync button for example. This might feel safer to them sometimes, esp.
+   * when shifting between offline and online a lot.
+   *
+   * TODO: move to sync.js
```
What about the TODO there? Didn't you just move the doc?

gregkare replied:
I moved this comment that was prefixed with `startSync` from the top of the file to the top of the `startSync` function. Is `TODO: move to sync.js` saying that the comment needs to be moved to the `sync.sync` function?

skddc replied:
I'm not sure what's unclear about the language there. But I don't see why it should be moved. I asked you what's with the TODO there, because you looked at the doc and apparently already moved it to the appropriate place.

skddc commented on src/eventhandling.js:
```diff
@@ -1,12 +1,10 @@
 
   /**
-   * Interface: eventhandling
+   * @interfacek
```
Typo with the k there?

skddc commented on src/indexeddb.js:
```diff
@@ -90,6 +83,9 @@
 
   IndexedDB.prototype = {
+    /**
+     * @todo Document
```
We use `TODO` comments instead of these currently.

gregkare replied:
I'll switch them back to `TODO`

skddc commented:
Awesome! Just added a couple of comments/questions.

gregkare commented:
I fixed the issues and removed the `TODO: move to sync.js` since it doesn't seem to make sense. I think this should be ready to merge. Do we want to generate docs for the other files or is that not needed?

skddc commented:
> Do we want to generate docs for the other files or is that not needed?

You mean for private APIs? I think we have all public functions covered now, or maybe something is missing?
